logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/asset/ScheduleNewAssetMessageUseCase.kt

```diff
@@ -109,66 +113,108 @@ internal class ScheduleNewAssetMessageUseCaseImpl(
-                senderUserId = userId,
-                senderClientId = currentClientId,
-                status = Message.Status.PENDING,
-                editStatus = Message.EditStatus.NotEdited,
-                expectsReadConfirmation = expectsReadConfirmation,
-                expirationData = expireAfter?.let { duration ->
-                    // normalize the duration in case it's 0 to null, so that the message is not expirable in that case
-                    if (duration == Duration.ZERO) null
-                    else Message.ExpirationData(expireAfter, Message.ExpirationData.SelfDeletionStatus.NotStarted)
-                },
-                isSelfMessage = true
-            )
-
-            // We persist the asset message right away so that it can be displayed on the conversation screen loading
-            persistMessage(message).onSuccess {
-                // We schedule the asset upload and return Either.Right(Unit) so later it's transformed to Success(message.id)
+                assetDataPath = assetDataPath,
+                assetDataSize = assetDataSize,
+                assetName = assetName,
+                assetMimeType = assetMimeType,
+                assetWidth = assetWidth,
+                assetHeight = assetHeight,
+                expireAfter = expireAfter,
+                expectsReadConfirmation = expectsReadConfirmation
+            ).onSuccess { (currentAssetMessageContent, message) ->
+                // We schedule the asset upload and return Either.Right so later it's transformed to Success(message.id)
                 scope.launch(dispatcher.io) {
                     uploadAssetAndUpdateMessage(currentAssetMessageContent, message, conversationId, expectsReadConfirmation)
+                        .onSuccess {
+                            // We delete asset added temporarily that was used to show the loading
+                            assetDataSource.deleteAssetLocally(currentAssetMessageContent.assetId.key)
+                        }
                 }
             }
         }.fold({
-            updateAssetMessageUploadStatus(Message.UploadStatus.FAILED_UPLOAD, conversationId, message.id)
             ScheduleNewAssetMessageResult.Failure(it)
-        }, {
+        }, { (_, message) ->
             ScheduleNewAssetMessageResult.Success(message.id)
         })
     }
 
+    private suspend fun persistInitiallyAssetAndMessage(
+        messageId: String,
+        conversationId: ConversationId,
+        assetDataPath: Path,
+        assetDataSize: Long,
+        assetName: String,
+        assetMimeType: String,
+        assetWidth: Int?,
+        assetHeight: Int?,
+        expireAfter: Duration?,
+        expectsReadConfirmation: Boolean
+    ): Either<CoreFailure, Pair<AssetMessageMetadata, Message.Regular>> = currentClientIdProvider().flatMap { currentClientId ->
+        // Create a temporary asset key and domain
+        val (generatedAssetUuid, tempAssetDomain) = uuid4().toString() to ""
```

Comment:
why we are passing empty domain?

Reply:
It's just temporary to store asset file locally before it's uploaded so the domain shouldn't matter and generated unique asset id is enough because we will be fetching such asset only locally until it's uploaded.